messaging/src/main/java/org/axonframework/commandhandling/distributed/AbstractRoutingStrategy.java

```diff
@@ -17,62 +17,99 @@
 package org.axonframework.commandhandling.distributed;
 
 import org.axonframework.commandhandling.CommandMessage;
-import org.axonframework.common.Assert;
+import org.axonframework.common.AxonConfigurationException;
 
-import java.util.concurrent.atomic.AtomicLong;
-
-import static java.lang.String.format;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
 
 /**
- * Abstract implementation of the RoutingStrategy interface that uses a policy to prescribe what happens when a routing
- * cannot be resolved.
+ * Abstract implementation of the {@link RoutingStrategy} interface that includes a fallback {@code RoutingStrategy}
+ * which prescribes what happens when routing cannot be resolved by this implementation.
  *
  * @author Allard Buijze
  * @since 2.0
  */
-public abstract class AbstractRoutingStrategy implements RoutingStrategy {
+public abstract class AbstractRoutingStrategy<B extends RoutingStrategy> implements RoutingStrategy {
```

Comment:
Why is the generic parameter here? It doesn't seem to be used anywhere.

Reply:
That's for the `protected AbstractRoutingStrategy(Builder<B> builder) { ... }` constructor.
And added,I intended to create a `public abstract B builder();` method, which I missed.